src/psyclone/tests/psyir/symbols/datatype_test.py

```diff
@@ -285,6 +286,7 @@ def test_arraytype():
     assert shape0.upper.value == "10"
     assert shape0.upper.datatype.intrinsic == ScalarType.Intrinsic.INTEGER
     assert shape0.upper.datatype.precision == ScalarType.Precision.UNDEFINED
+    assert array_type.datatype == scalar_type
```

Comment:
See discussion above, I think this test is rather useful while porting more and more code, where me might end up with inconsistent types. Once all is done properly (e.g. we consistently declare all symbols in the initialisation, and when using the symbols, we can assume they are all correct), this can likely be removed, but till then it's rather important IMHO.

I've added TODOs for #1857 for these lines.